integration/common/kubeclient.go

```diff
@@ -22,3 +31,95 @@ func GetKubeClientset() (*kubernetes.Clientset, string, error) {
 	clientset, err := kubernetes.NewForConfig(restClientConfig)
 	return clientset, ns, err
 }
+
+func RunSimpleBuildImageAsPod(ctx context.Context, name, imageName, namespace string, clientset *kubernetes.Clientset) error {
+	podClient := clientset.CoreV1().Pods(namespace)
+	eventClient := clientset.CoreV1().Events(namespace)
+	logrus.Infof("starting pod %s for image: %s", name, imageName)
+	// Start the pod
+	pod, err := podClient.Create(ctx,
+		&v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:            name,
+						Image:           imageName,
+						Command:         []string{"sleep", "60"},
+						ImagePullPolicy: v1.PullNever,
+					},
+				},
+			},
+		},
+		metav1.CreateOptions{},
+	)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		err := podClient.Delete(ctx, pod.Name, metav1.DeleteOptions{})
+		logrus.Warnf("failed to clean up pod %s: %s", pod.Name, err)
+	}()
+
+	logrus.Infof("waiting for pod to start...")
+	// Wait for it to get started, and make sure it isn't complaining about image not being found
+	// TODO - multi-node test clusters will need some refinement here if we wind up not scaling the builder up in some scenarios
+	var refUID *string
+	var refKind *string
+	reportedEvents := map[string]interface{}{}
+
+	for try := 0; try < 100; try++ {
```

Comment:
do you want to shove 100 into a const?

Reply:
Good suggestion.  But, actually, now that I'm thinking about this chunk a bit more, let me see if I can DRY this out with the origin code I copied it from in the mainline driver code.

Reply:
Actually, on second thought, I refactored it a fair bit, so DRYing these two code paths out will require non-trivial changes to the already working main-line code.  I'll defer this for now so we can get this resolved and put a comment in there for a breadcrumb.